packages/brisa/src/utils/file-system-router/index.ts

```diff
@@ -131,6 +131,12 @@ function getParamsAndQuery(route: string, pathname: string, url: URL) {
   );
 
   const query = { ...params, ...Object.fromEntries(url.searchParams) };
```

Comment:
After your change, does not make sense to use `Object.fromEntries(url.searchParams)` . Maybe we can move your code to some function like:

```ts
function getURLParams(url: URL) {
  const params = {}

  for (const key of url.searchParams.keys()) {
    params[key] = url.searchParams.getAll(key);

    if (params[key].length === 1) {
      params[key] = params[key][0];
    }
  }

  return params;
}
```

And here:

```suggestion
  const query = { ...params, ...getURLParams(url) };
```